src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesUpdateQueueService.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.incentives
+
+import com.amazonaws.services.sqs.model.SendMessageRequest
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.listeners.SQSMessage
+import uk.gov.justice.hmpps.sqs.HmppsQueue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+
+@Service
+class IncentivesUpdateQueueService(
+  private val hmppsQueueService: HmppsQueueService,
+  private val telemetryClient: TelemetryClient,
+  private val objectMapper: ObjectMapper,
+) {
+  private val queue by lazy { hmppsQueueService.findByQueueId("incentive") as HmppsQueue }
+  private val sqsClient by lazy { queue.sqsClient }
+  private val queueUrl by lazy { queue.queueUrl }
+
+  fun sendMessage(context: IncentiveContext) {
+    val sqsMessage = SQSMessage(
+      Type = "RETRY",
+      Message = objectMapper.writeValueAsString(context),
+      MessageId = "retry-${context.incentiveId}"
+    )
+    val result = sqsClient.sendMessage(
+      SendMessageRequest(queueUrl, objectMapper.writeValueAsString(sqsMessage))
+    )
+
+    telemetryClient.trackEvent(
+      "create-map-queue-retry",
```

Comment:
does this custom event name need to be subject (incentive) specific?

Reply:
Good idea, will add for next time